suite/spdz/src/main/java/dk/alexandra/fresco/suite/spdz/gates/SpdzMacCheckProtocol.java

```diff
@@ -49,16 +55,21 @@ public SpdzMacCheckProtocol(SecureRandom rand, MessageDigest digest, SpdzStorage
         BigInteger s = new BigInteger(modulus.bitLength(), rand).mod(modulus);
         SpdzCommitment commitment = new SpdzCommitment(digest, s, rand);
         Map<Integer, BigInteger> comms = new HashMap<>();
-        SpdzCommitProtocol comm = new SpdzCommitProtocol(commitment, comms);
-        SpdzOpenCommitProtocol open = new SpdzOpenCommitProtocol(commitment, comms, commitments);
+        comm = new SpdzCommitProtocol(commitment, comms);
+        openComm = new SpdzOpenCommitProtocol(commitment, comms, commitments);
 
-        pp = new SequentialProtocolProducer(
-            new SingleProtocolProducer<>(comm),
-            new SingleProtocolProducer<>(open));
+        pp = new SequentialProtocolProducer(new SingleProtocolProducer<>(comm),
+            new SingleProtocolProducer<>(openComm));
       } else if (round == 1) {
-        BigInteger alpha = storage.getSSK();
+        if (!comm.out()) {
+          throw new MPCException(
```

Comment:
This should be Malicious if it in fact is malicious

Reply:
What do you mean? The exception? Isn't this only within the mascot branch currently?

Reply:
You are right, sorry